devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy

```diff
@@ -79,23 +94,127 @@ metadata {
+        input "configParam${param.num}", "enum", title: "${param.name}:", required: false, defaultValue: "${param.value}", options: param.options
+    }
+}
+
+private addChildButton() {
+    log.warn "Creating Button Device"
+    def child = addChildDevice(
+            "sky-nie",
+            "Child Button",
+            "${device.deviceNetworkId}-2",
+            device.getHub().getId(),
+            [
+                completedSetup: true,
+                isComponent: false,
+                label: "plugButton",
+                componentLabel: "${device.displayName[0..-8]} Button"
+            ]
+    )
+    child?.sendEvent(name:"supportedButtonValues", value:JsonOutput.toJson(["pushed", "down", "down_2x", "up", "up_2x"]), displayed:false)
+    child?.sendEvent(name:"numberOfButtons", value:1, displayed:false)
+    sendButtonEvent("pushed")
+    return child
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+    logTrace "CentralSceneNotification: ${cmd}"
+    if (state.lastSequenceNumber != cmd.sequenceNumber) {
+        state.lastSequenceNumber = cmd.sequenceNumber
+        logTrace "${cmd}"
+        def paddle = (cmd.sceneNumber == 1) ? "down" : "up"
+        def btnVal
+        switch (cmd.keyAttributes) {
+            case 0:
+                btnVal = paddle
+                break
+            case 3:
+                btnVal = paddle + "_2x"
+                break
+        }
+
+        if (btnVal) {
+            sendButtonEvent(btnVal)
         }
     }
+    return []
+}
+
+private sendButtonEvent(value) {
+    childDevices[0].sendEvent(name: "button", value: value, data:[buttonNumber: 1], isStateChange: true)
+}
+
+def ping() {
+    logDebug "ping()..."
+    return [ switchMultilevelGetCmd() ]
+}
+
+def refresh() {
+    logDebug "refresh()..."
+    refreshSyncStatus()
+    return [ switchMultilevelGetCmd() ]
+}
+
+private switchMultilevelGetCmd() {
+    return secureCmd(zwave.switchMultilevelV3.switchMultilevelGet())
 }
 
 def installed() {
     logDebug "installed()..."
+    try {
+        def child = addChildButton()
+        child?.sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+    } catch (ex) {
+        log.error("Unable to create button device because the 'Child Button' DTH is not installed",ex)
+    }
     sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
-    state.refreshConfig = true
+}
+
+def uninstalled() {
+    logger("debug", "uninstalled()")
+    removeChildButton(childDevices[0])
+}
+
+private removeChildButton(child) {
+    try {
+        log.warn "Removing ${child.displayName}} "
+        deleteChildDevice(child.deviceNetworkId)
+    } catch (ex) {
+        log.error("Unable to remove ${child.displayName}!  Make sure that the device is not being used by any SmartApps.", ex)
+    }
 }
```

Comment:
Since it's only called once in uninstall() - Is this function still needed here ?
I think child devices should be removed automatically when the main device is removed.

Reply:
oh, I'm not sure here. I wrote it by referring to the DTHs in SmartThingsCommunity. I found that some DTHs called deleteChildDevice in uninstalled, but some did not. I can't track uninstalled calls.
If the DTH is running in a separate virtual machine, it is not need to call "uninstalled()", because uninstalling the device is actually destroying the virtual machine. Otherwise,  it is  need to call "uninstalled()" to delete the child device.

Reply:
You might remove both functions: `uninstalled()` and `removeChildButton()` in this case.

Reply:
OK! I removed them.